### DISCUSSION TITLE
Melakukkan Padding di Time Series dengan interval hour

### DISCUSSION THREAD

**Discussion by jeremiaww** [teamalgoritma/community#148] [Q&A] [answered]
Izin bertanya terkait error pada padding, yang ingin dilakukan adalah melakukan padding pada datetime dengan interval hour.

<img width="1026" alt="image" src="https://user-images.githubusercontent.com/99194718/175191548-d00424d2-dc9d-4ac3-8838-d2b8991eb10a.png">

Terima kasih.

**Comment by kevbow** [comment]
Halo Pak Jeremia, 

Sepertinya ada masalah ketika preparasi data, seperti error yang tertulis bahwa data masih berbentuk grouping. Mungkin bisa dicek kembali ketika melakukan agregasi data. 
Demikian dan terima kasih.

**Comment by muhrisman** [comment, marked as answer]
Halo Pak Jeremia,

Sebelumnya mau melakukan konfirmasi dulu, untuk padding yang sedang dilakukan apakah sudah melalui tahap wrangling yang lain? Misalkan ada pemberian grouping pada suatu variabel. Jika terdapat wrangling seperti itu, bapak bisa mengecek kembali agregasi data yang dilakukan.

Sebagai saran, untuk kasus scotty ini saya ada saran terkait padding kasus data hour.
1.  Pertama bapak buat dulu object time range dengan menggunakan min dan max. Variabel yang dipake di sini adalah variabel waktu dalam data. 
2. Bapak lakukan eksplorasi menggunakan fungsi make_difftime untuk mencari rentang waktu. Isi datri fungsi ini adalah object min max yang bapak baut di tahap pertama.
3. Bapak lakukan agregasi pada data frame. Karena kasus scotty ini dibagi menjadi beberapa daerah, bapak lakukan grouping sesuai dengan variabel daerah yang ada di dataset.
4. Bapak lakukan pading berdasarkan step nomor 2.

Semoga membantu :)
